app/src/main/java/com/internship/move/di/appModule.kt

```diff
@@ -1,26 +1,64 @@
 package com.internship.move.di
 
-import com.internship.move.repository.Repository
+import com.internship.move.BuildConfig
+import com.internship.move.network.ServiceApi
+import com.internship.move.repository.UserRepository
 import com.internship.move.ui.authentication.AuthenticationViewModel
 import com.internship.move.ui.home.MainViewModel
 import com.internship.move.ui.onboarding.OnboardingViewModel
 import com.internship.move.ui.splash.SplashViewModel
 import com.internship.move.utils.InternalStorageManager
+import com.squareup.moshi.Moshi
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
 import org.koin.android.ext.koin.androidContext
 import org.koin.androidx.viewmodel.dsl.viewModel
 import org.koin.dsl.module
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import java.util.concurrent.TimeUnit
 
 val viewModels = module {
-    viewModel { SplashViewModel(repo = get()) }
-    viewModel { OnboardingViewModel(repo = get()) }
-    viewModel { AuthenticationViewModel(repo = get()) }
-    viewModel { MainViewModel(repo = get()) }
+    viewModel { SplashViewModel(internalStorageManager = get()) }
+    viewModel { OnboardingViewModel(internalStorageManager = get()) }
+    viewModel { AuthenticationViewModel(repo = get(), internalStorageManager = get()) }
+    viewModel { MainViewModel(internalStorageManager = get()) }
 }
 
 val repositories = module {
-    single { Repository(internalStorageManager = get()) }
+    single { UserRepository(api = get()) }
+}
+
+val services = module {
+    single { getMoshi() }
+    single { getOkHttpClient() }
+    single { getRetrofit(moshi = get(), httpClient = get()) }
+    single { getService(retrofit = get()) }
 }
 
 val storage = module {
     single { InternalStorageManager(androidContext()) }
-}
\ No newline at end of file
+}
+
+fun getMoshi(): Moshi = Moshi.Builder().build()
+
+fun getOkHttpClient(): OkHttpClient {
+    val logging = HttpLoggingInterceptor()
+    logging.level = HttpLoggingInterceptor.Level.BODY
+    return OkHttpClient.Builder()
+        .addInterceptor(logging)
+        .connectTimeout(10, TimeUnit.SECONDS)
```

Comment:
Extract magic numbers to appropriately named constants
I believe you can now create a Constants file, with a `Constants` object that has other specific objects in it to separate things a bit